### PR TITLE
Dynamic height text area for chat input

### PR DESCRIPTION
**Pull request by GrahamS-Quartech**
### Motivation and Context
When constructing a large prompt, it can be difficult to skim your own input with only two lines of text present at a time in the TextArea. This change dynamically adjusts the height based on the text content.

### Description

* Using the textArea ref, dynamically adjust the height based on the content's scrollHeight in the onChange handler.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [Contribution Guidelines](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/chat-copilot/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Review by jdtoombs — APPROVED** [review]

**Comment by jdtoombs** [comment]
Looks good - I know ChatGPT does it where it has a `max-height` so it doesn't ever get too crazy, but that could be a later thing!

**Comment by wdavidcamden** [comment]
If we enter Dean's crazy long prompt for the IAF specialization, does it still work?

**Review by wdavidcamden — CHANGES_REQUESTED** [review]
If we enter Dean's crazy long prompt for the IAF specialization, does it still work?

**Comment by GrahamS-Quartech** [comment]
image

Currently it maxes out at this height, then you can scroll around from there. Wasn't sure if we wanted this thing completely taking over the screen or not.

**Comment by wdavidcamden** [comment]
> image
> 
> Currently it maxes out at this height, then you can scroll around from there. Wasn't sure if we wanted this thing completely taking over the screen or not.

Can we add a scrollbar? then it looks good to me :)

File: webapp/src/components/chat/ChatInput.tsx [hunk 1]
```diff
@@ -311,6 +314,10 @@ export const ChatInput: React.FC<ChatInputProps> = ({ disabled, isDraggingOver,
                         }
 
                         setInput(data.value);
+                        if (textAreaRef.current) {
+                            textAreaRef.current.style.height = 'auto';
+                            textAreaRef.current.style.height = `${textAreaRef.current.scrollHeight}px`;
```
Comment (JTraill) [inline webapp/src/components/chat/ChatInput.tsx:319]: `textAreaRef.current.style.height` is being set twice here

**Review by JTraill — APPROVED** [review]

File: webapp/src/components/chat/ChatInput.tsx (hunk 1 above)
Comment (GrahamS-Quartech) [reply webapp/src/components/chat/ChatInput.tsx:319]: This is intentional, try commenting out either line and you will see why.

**Comment by GrahamS-Quartech** [comment]
> Can we add a scrollbar? then it looks good to me :)

There is a scrollbar when the input has focus, just not shown in that screenshot.

**Pull request merged**